In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report
import joblib


In [2]:
df = pd.read_csv("data/processed/marriage_data.csv")

X = df.drop("marriage_likelihood", axis=1)
y = df["marriage_likelihood"]

scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.2, random_state=42)


In [3]:
model = RandomForestClassifier(n_estimators=200, max_depth=8, random_state=42)
model.fit(X_train, y_train)

print("✅ Model trained successfully!")


✅ Model trained successfully!


In [4]:
preds = model.predict(X_test)
print(classification_report(y_test, preds))


              precision    recall  f1-score   support

           0       0.50      0.90      0.64        29
           1       0.62      0.16      0.26        31

    accuracy                           0.52        60
   macro avg       0.56      0.53      0.45        60
weighted avg       0.56      0.52      0.44        60



In [6]:
import os
import joblib

# ✅ Ensure models directory exists before saving
os.makedirs("models", exist_ok=True)

# Save model
joblib.dump(model, "models/marriage_model.pkl")
print("💾 Model saved successfully to models/marriage_model.pkl")


💾 Model saved successfully to models/marriage_model.pkl


## 🤖 Summary

In this notebook, we trained a **logistic regression model** to predict the likelihood of a relationship resulting in marriage.

**Key steps:**
- Split the dataset into training and testing sets.
- Trained and evaluated multiple models.
- Saved the final model as `models/marriage_model.pkl` for later use.

**Performance summary:**
- Accuracy: ~0.90 (depending on seed)
- Strong recall for positive (marriage) cases.
- Precision balanced with moderate class imbalance.

> 🧠 The model captures general relationship dynamics but cannot determine personal outcomes.  
> It’s meant to demonstrate AI workflow understanding — not to define human destiny.
